t/instant_answer_basics.t

```diff
@@ -0,0 +1,137 @@
+use strict;
+use warnings;
+
+# Database setup
+BEGIN {
+    $ENV{DDGC_DB_DSN} = 'dbi:SQLite:dbname=ddgc_test.db';
+}
+
+use Test::More;
+use t::lib::DDGC::TestUtils;
+use HTTP::Request::Common;
+use Plack::Test;
+use Plack::Builder;
+use Plack::Session::State::Cookie;
+use Plack::Session::Store::File;
+use File::Temp qw/ tempdir /;
+use JSON::MaybeXS qw/:all/;
+use URI;
+
+use DDGC;
+use DDGC::Web;
+
+my $d = DDGC->new;
+t::lib::DDGC::TestUtils::deploy( undef, $d->db );
+
+# Application setup and creation
+my $app = builder {
+    enable 'Session',
+        store => Plack::Session::Store::File->new(
+            dir => tempdir,
+        ),
+        state => Plack::Session::State::Cookie->new(
+            secure => 0,
+            httponly => 1,
+            expires => 21600,
+            session_key => 'ddgc_session',
+        );
+    mount '/testutils' => t::lib::DDGC::TestUtils->to_app;
+    mount '/' => DDGC::Web->new->psgi_app;
+};
+
+# Tests - assert things here
+test_psgi $app => sub {
+    my ( $cb ) = @_;
+
+    my $get_repo_json = sub {
+        my $opts = shift;
+        my $repo = $opts->{repo} || 'all';
+        my $u = URI->new("/ia/repo/$repo/json");
+        $u->query_form( all_milestones => $opts->{all_milestones} || 0 );
+
+        my $ia_repo_json_request = $cb->( GET $u->canonical );
+        ok( $ia_repo_json_request->is_success, "Can retrieve $repo JSON" );
+        return decode_json( $ia_repo_json_request->decoded_content );
+    };
+
+    my $get_action_token = sub {
+        my $action_token = $cb->(
```

Comment:
@MariagraziaAlastra The reason I went with coderefs like this is stuff in the `test_psgi` scope (like `$cb`) is useful, so we don't have to pass around arbitrary stuff from this scope to external subs.

Thoughts?